core/tests/performance_framework/perf_services.py

```diff
@@ -130,12 +134,14 @@ def get_page_metrics_from_cached_session(self, page_url):
         return perf_domain.PageSessionMetrics(
             page_session_stats=har_dict, page_session_timings=None)
 
-    def get_page_timings_for_url(self, page_url):
+    def get_page_timings_for_url(self, page_url, func_call):
         """Returns a PageSessionMetrics domain object initialized using
         page load timings for a page URL.
         """
         with Xvfb() as _:
```

Comment:
I'm not keen on the callback pattern being used here (with ``func_call``). That being said, does Xvfb need to be allocated for all of these function calls? Does it only need to exist for the ``setup_driver`` bit, or does it also need to be here for ``driver.get(page_url)`` and ``self._wait_until_page_load_is_finished()``?